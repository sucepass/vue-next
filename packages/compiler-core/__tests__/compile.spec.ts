// Integration tests for parser + transform + codegen
test.todo('compile')
