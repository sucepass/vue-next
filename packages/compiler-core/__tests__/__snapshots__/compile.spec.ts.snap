// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler: integration tests function mode 1`] = `
"const _Vue = Vue

return function render() {
  with (this) {
    const { toString: _toString, openBlock: _openBlock, createVNode: _createVNode, createBlock: _createBlock, Comment: _Comment, Fragment: _Fragment, renderList: _renderList } = _Vue
    
    return (_openBlock(), _createBlock(\\"div\\", {
      id: \\"foo\\",
      class: bar.baz
    }, [
      _toString(world.burn()),
      (_openBlock(), ok
        ? _createBlock(\\"div\\", { key: 0 }, \\"yes\\")
        : _createBlock(_Fragment, { key: 1 }, [\\"no\\"])),
      (_openBlock(), _createBlock(_Fragment, null, _renderList(list, (value, index) => {
        return (_openBlock(), _createBlock(\\"div\\", null, [
          _createVNode(\\"span\\", null, _toString(value + index), 1 /* TEXT */)
        ]))
      }), 128 /* UNKEYED_FRAGMENT */))
    ], 2 /* CLASS */))
  }
}"
`;

exports[`compiler: integration tests function mode w/ prefixIdentifiers: true 1`] = `
"const { toString, openBlock, createVNode, createBlock, Comment, Fragment, renderList } = Vue

return function render() {
  const _ctx = this
  return (openBlock(), createBlock(\\"div\\", {
    id: \\"foo\\",
    class: _ctx.bar.baz
  }, [
    toString(_ctx.world.burn()),
    (openBlock(), (_ctx.ok)
      ? createBlock(\\"div\\", { key: 0 }, \\"yes\\")
      : createBlock(Fragment, { key: 1 }, [\\"no\\"])),
    (openBlock(), createBlock(Fragment, null, renderList(_ctx.list, (value, index) => {
      return (openBlock(), createBlock(\\"div\\", null, [
        createVNode(\\"span\\", null, toString(value + index), 1 /* TEXT */)
      ]))
    }), 128 /* UNKEYED_FRAGMENT */))
  ], 2 /* CLASS */))
}"
`;

exports[`compiler: integration tests module mode 1`] = `
"import { toString, openBlock, createVNode, createBlock, Comment, Fragment, renderList } from \\"vue\\"

export default function render() {
  const _ctx = this
  return (openBlock(), createBlock(\\"div\\", {
    id: \\"foo\\",
    class: _ctx.bar.baz
  }, [
    toString(_ctx.world.burn()),
    (openBlock(), (_ctx.ok)
      ? createBlock(\\"div\\", { key: 0 }, \\"yes\\")
      : createBlock(Fragment, { key: 1 }, [\\"no\\"])),
    (openBlock(), createBlock(Fragment, null, renderList(_ctx.list, (value, index) => {
      return (openBlock(), createBlock(\\"div\\", null, [
        createVNode(\\"span\\", null, toString(value + index), 1 /* TEXT */)
      ]))
    }), 128 /* UNKEYED_FRAGMENT */))
  ], 2 /* CLASS */))
}"
`;
